h2o-core/src/main/java/water/JettyHTTPD.java

```diff
@@ -145,7 +145,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         try { Thread.sleep(100); }
         catch (Exception ignore) {}
       }
-      setCommonResponseHttpHeaders(response, isXhrRequest(baseRequest));
+      setCommonResponseHttpHeaders(response, isXhrRequest(request));
```

Comment:
When can the `request` be null? If it is null the method should not be called - the handling should be on the caller's side.

Reply:
For example in test `water/api/CustomHttpFilterTest.java:103` there is null `baseRequest`. AFAIK this is the only reasonable situation with the null `baseRequest`.